src/Compilers/Core/MSBuildTaskTests/GenerateMSBuildEditorConfigTests.cs

```diff
@@ -351,5 +353,29 @@ public void ItemMetadataPathIsAdjustedOnWindows()
 build_metadata.Compile.ToRetrieve = abc123
 ", result);
         }
+
+        [Fact]
+        public void ConfigFileCanBeWrittenToDisk()
+        {
+            ITaskItem property1 = MSBuildUtil.CreateTaskItem("Property1", new Dictionary<string, string> { { "Value", "abc123" } });
+            ITaskItem property2 = MSBuildUtil.CreateTaskItem("Property2", new Dictionary<string, string> { { "Value", "def456" } });
+
+            var fileName = Path.Combine(TempRoot.Root, "ConfigFileCanBeWrittenToDisk.GenerateMSBuildEditorConfig.editorconfig");
+
+            GenerateMSBuildEditorConfig configTask = new GenerateMSBuildEditorConfig()
+            {
+                PropertyItems = new[] { property1, property2 },
+                FileName = new TaskItem(fileName)
+            };
+            configTask.Execute();
+
+            var expectedContents = @"is_global = true
+build_property.Property1 = abc123
+build_property.Property2 = def456
+";
+
+            Assert.True(File.Exists(fileName));
+            Assert.Equal(expectedContents, File.ReadAllText(fileName));
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider adding `Assert.False(configTask.WriteMSBuildEditorConfig());`

Reply:
I was hoping to test that `WriteMSBuildEditorConfig()` did not update the file if the contents were already correct, but it looks like It looks the method returns `true` in either case.

Should we add an `out bool updated` parameter that indicated whether the file was modified, and check that value?